clang/include/clang/Sema/Sema.h

```diff
@@ -4024,6 +4024,19 @@ class Sema {
     void disable() { Active = false; }
   };
 
+  // Checked C: Perform semantic analysis on a where clause.
+  WhereClause *ActOnWhereClause(SourceLocation Loc);
+
+  // Checked C: Perform semantic analysis on a where clause bounds fact.
+  WhereClauseFact *ActOnBoundsFact(IdentifierInfo *Id, Expr *E,
+                                   Scope *CurScope, SourceLocation Loc);
+
+  // Checked C: Perform semantic analysis on a where clause relop fact.
+  WhereClauseFact *ActOnRelopFact(Expr *E, SourceLocation Loc);
+
+  // Ignore value preserving casts.
```

Comment:
This `IgnoreCasts` method also seems to ignore `LValueToRValue` casts, which I don't think we consider value-preserving.

Also, `Expr` has other cast-ignoring methods - should this be an `Expr` method rather than `Sema`?